build/document-extractor.js

```diff
@@ -236,6 +236,60 @@ function _addSingleSectionBCD($) {
   if (data === undefined) {
     return [];
   }
+
+  // First extract a map of all release data, keyed by (normalized) browser
+  // name and the versions.
+  // You'll have a map that looks like this:
+  //
+  //   'chrome_android': {
+  //      '28': {
+  //        release_data: '2012-06-01',
+  //        release_notes: '...',
+  //        ...
+  //
+  // The reason we extract this to a locally scoped map, is so we can
+  // use it to augment the `__compat` blocks for the latest version
+  // when (if known) it was added.
+  const browserReleaseData = new Map();
+  Object.entries(browsers).forEach(([name, browser]) => {
+    const releaseData = new Map();
+    for (const [version, data] of Object.entries(browser.releases || [])) {
+      if (data) {
+        releaseData.set(version, data);
+      }
+    }
+    browserReleaseData.set(name, releaseData);
+  });
```

Comment:
nit: If you're into map/filter as much as I am, you'll find this more readable:

```javascript
const browserReleaseData = new Map(
  Object.entries(browsers).map(([name, browser]) => [
    name,
    new Map(
      Object.entries(browser.releases || []).filter(([, data]) => !!data)
    ),
  ])
);
```

In any case I'd vote for consistently using for...of/forEach (and I prefer the former, as it allows for early returns).

Reply:
Yay for consistency! Yay for code that is "dumber" but reads without feeling like an exercise. And Yay for code that makes it possible to inject a little `console.log` if you have to :)

Reply:
I actually think the functional version is dumber because it's more declarative and doesn't require you to maintain state in your head while reading it. Fortunately `log` is an expression and can also be used like this: `Object.entries(console.log(browser.releases) || browser.releases || [])`.

So for me it is strictly better, but I know opinions diverge on this topic so I marked it as a nit.